packages/ui-components/src/components/action-button/action-button.tsx

```diff
@@ -33,11 +35,15 @@ export class KvActionButton implements IButton, IButtonEvents, IAnchor {
 	/** @inheritdoc */
 	@Event() blurButton: EventEmitter<FocusEvent>;
 
+	@State() loadingState: boolean = false;
```

Comment:
Dont forget that since you are using state you will need to sync this variable with the prop `loading` otherwise, no visible change will be seen.

Maybe a componentWillLoad and then a Watch to sync the prop value. However, I believe here a state is not necessary. What do you think?

Reply:
I agree with @GoncaloCamaz, this is best as a prop as you seem to be tying the loading logic to the click of the button and from what I could gather with this approach you'll not be able to unset the loading styling from the parent, what we want is to be able to set a prop that defines if the button is loading according to some operation state.